core/app/models/spree/product.rb

```diff
@@ -110,11 +110,7 @@ def has_variants?
     end
 
     def tax_category
-      if self[:tax_category_id].nil?
-        TaxCategory.where(is_default: true).first
-      else
-        TaxCategory.find(self[:tax_category_id])
-      end
+      super || TaxCategory.find_by(is_default: true)
```

Comment:
As I understand, Globalize not works fine with `find_by`, but in that case Globalize is not used, right?